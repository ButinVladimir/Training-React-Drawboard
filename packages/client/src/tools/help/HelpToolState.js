import BaseToolState from '../BaseToolState';

class HelpToolState extends BaseToolState {
  constructor() {
    super();

    this.anchorX = 0;
    this.anchorY = 0;

    this.clientX = 0;
    this.clientY = 0;

    this.speed = 0;
  }

  restore() { // eslint-disable-line class-methods-use-this,no-unused-vars
    throw new Error('Method is not implemented');
  }

  setDefaultSpeed() { // eslint-disable-line class-methods-use-this,no-unused-vars
    throw new Error('Method is not implemented');
  }

  onMouseDown(event) {
    this.anchorX = event.clientX;
    this.anchorY = event.clientY;

    this.clientX = event.clientX;
    this.clientY = event.clientY;
  }

  onMouseMove(event) {
    this.clientX = event.clientX;
    this.clientY = event.clientY;
  }

  onMouseUp() {
    this.clientX = this.anchorX;
    this.clientY = this.anchorY;
  }

  serialize() { // eslint-disable-line class-methods-use-this
    return null;
  }

  static deserialize() {
    return null;
  }
}

export default HelpToolState;
