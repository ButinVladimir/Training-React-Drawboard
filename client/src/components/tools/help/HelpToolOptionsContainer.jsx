import React, { Component } from 'react';
import PropTypes from 'prop-types';
import HelpToolOptions from './HelpToolOptions';
import HelpToolState from '../../../tools/help/HelpToolState';

class HelpToolOptionsContainer extends Component {
  constructor(props) {
    super(props);

    this.onRestore = this.onRestore.bind(this);
    this.onChangeSpeed = this.onChangeSpeed.bind(this);
    this.onSetDefaultSpeed = this.onSetDefaultSpeed.bind(this);

    const { toolState } = this.props;
    const parsedSpeed = parseFloat(toolState.speed);

    this.state = {
      speed: parsedSpeed,
      parsedSpeed,
    };
  }

  onRestore() {
    const { onRestore } = this.props;

    onRestore();
  }

  onChangeSpeed(speed) {
    const { toolState } = this.props;
    const parsedSpeed = parseFloat(speed);

    this.setState({ speed, parsedSpeed });
    toolState.speed = parsedSpeed;
  }

  onSetDefaultSpeed() {
    const { toolState } = this.props;

    toolState.setDefaultSpeed();
    this.setState({
      speed: toolState.speed,
      parsedSpeed: toolState.speed,
    });
  }

  render() {
    const { speed, parsedSpeed } = this.state;

    return (
      <HelpToolOptions
        speed={speed}
        parsedSpeed={parsedSpeed}
        onChangeSpeed={this.onChangeSpeed}
        onSetDefaultSpeed={this.onSetDefaultSpeed}
        onRestore={this.onRestore}
      />
    );
  }
}

HelpToolOptionsContainer.propTypes = {
  toolState: PropTypes.instanceOf(HelpToolState).isRequired,
  onRestore: PropTypes.func.isRequired,
};

export default HelpToolOptionsContainer;
