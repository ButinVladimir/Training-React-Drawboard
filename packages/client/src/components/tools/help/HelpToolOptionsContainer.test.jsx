import React from 'react';
import { shallow, mount } from 'enzyme';
import HelpToolOptionsContainer from './HelpToolOptionsContainer';
import HelpToolState from '../../../tools/help/HelpToolState';

describe('HelpToolOptionsContainer', () => {
  it('renders properly', () => {
    const speed = 3;
    const toolState = new HelpToolState();
    Object.assign(toolState, { speed });

    const component = shallow(<HelpToolOptionsContainer
      toolState={toolState}
      onRestore={() => {}}
    />);
    const componentState = component.state();

    expect(component).toMatchSnapshot();
    expect(componentState).toEqual({
      speed,
      parsedSpeed: speed,
    });
  });

  it('handles onChangeSpeed event', () => {
    const toolState = new HelpToolState();
    const component = mount(<HelpToolOptionsContainer
      toolState={toolState}
      onRestore={() => {}}
    />);
    const speed = '5';
    const parsedSpeed = 5;

    component.find('input[name="speed"]').simulate('change', { target: { value: speed } });
    const componentState = component.state();

    expect(toolState.speed).toBe(parsedSpeed);
    expect(componentState).toMatchObject({
      speed,
      parsedSpeed,
    });
  });

  it('handles onSetDefaultSpeed event', () => {
    const speed = 4;
    const toolState = new HelpToolState();
    toolState.setDefaultSpeed = jest.fn(() => { toolState.speed = speed; });
    const component = mount(<HelpToolOptionsContainer
      toolState={toolState}
      onRestore={() => {}}
    />);

    component.find('button[name="set-default-speed"]').simulate('click');
    const componentState = component.state();

    expect(toolState.setDefaultSpeed).toHaveBeenCalled();
    expect(toolState.speed).toBe(speed);
    expect(componentState).toMatchObject({
      speed,
      parsedSpeed: speed,
    });
  });
});
