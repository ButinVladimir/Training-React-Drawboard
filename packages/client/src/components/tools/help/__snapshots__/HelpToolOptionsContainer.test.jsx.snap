// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpToolOptionsContainer renders properly 1`] = `
<HelpToolOptions
  onChangeSpeed={[Function]}
  onRestore={[Function]}
  onSetDefaultSpeed={[Function]}
  parsedSpeed={3}
  speed={3}
/>
`;
